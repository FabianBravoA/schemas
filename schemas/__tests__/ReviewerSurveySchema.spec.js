const mongoose = require('mongoose/browser');
const ReviewerSurveySchema = require('../ReviewerSurveySchema')(mongoose);

describe('ReviewerSurveySchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ReviewerSurveySchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should fail validation when multiple-choice question is missing options', () => {
    const doc = new mongoose.Document({
      questions: [{
        type: 'multiple-choice',
      }],
    }, ReviewerSurveySchema);
    return doc.validate()
      .catch(err => expect(err.message).toMatchSnapshot());
  });

  it('should fail validation when question is missing slug', () => {
    const doc = new mongoose.Document({
      questions: [{
        type: 'multiple-choice',
        options: 4,
      }],
    }, ReviewerSurveySchema);
    return doc.validate()
      .catch(err => expect(err.message).toMatchSnapshot());
  });

  it('should validate good reviewer survey', () => {
    const doc = new mongoose.Document({
      questions: [{
        slug: 'foo',
        type: 'multiple-choice',
        options: 4,
      }],
    }, ReviewerSurveySchema);
    return doc.validate();
  });
});
