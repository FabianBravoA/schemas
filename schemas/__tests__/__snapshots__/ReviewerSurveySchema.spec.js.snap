// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewerSurveySchema should fail validation when fields missing 1`] = `
Object {
  "questions": [ValidatorError: Path \`questions\` is required.],
}
`;

exports[`ReviewerSurveySchema should fail validation when multiple-choice question is missing options 1`] = `"Validation failed: questions: Options is required when type is multiple-choice"`;

exports[`ReviewerSurveySchema should fail validation when question is missing slug 1`] = `"Validation failed: questions.0.slug: Path \`slug\` is required."`;
